### PR TITLE
feat(swaps): add price impact warning

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

This PR:
- fix swaps not working because of `toFormat` in inputs by disabling formatting for now
- refactor swappable tokens logic so that sorting is included in the selector
- add price impact warning threshold remote config
- add price impact calculation when exchange rate changes
- show price impact warning when price impact exceeds 4%
- hide price impact warning on swap amount update / new (low price impact) exchange rate fetched
- add analytics

### Test plan

Added unit tests.

https://github.com/valora-inc/wallet/assets/20150449/d30e761e-3b4d-49c3-a201-530b7a2435ed

https://github.com/valora-inc/wallet/assets/20150449/5020d4ec-7052-4b67-a62b-86692c38518b

### Related issues

- Fixes RET-745

### Backwards compatibility

Y

### CODE REVIEW

File: src/tokens/selectors.ts [hunk 1]
```diff
@@ -124,8 +124,23 @@ export const celoAddressSelector = createSelector(coreTokensSelector, (tokens) =
   return tokens.find((tokenInfo) => tokenInfo.symbol === 'CELO')?.address
 })
 
+function tokenCompareByUsdBalanceThenByAlphabetical(token1: TokenBalance, token2: TokenBalance) {
```
Comment (kathaypacific) [inline src/tokens/selectors.ts:127]: i didn't change this, just moved from the component to here

File: src/swap/SwapScreen.tsx [hunk 2]
```diff
@@ -75,12 +59,19 @@ export function SwapScreenSection({ showDrawerTopNav }: { showDrawerTopNav: bool
...
   )
 
-  const supportedTokens = useSelector(
-    swappingNonNativeTokensEnabled ? swappableTokensSelector : coreTokensSelector
-  )
+  // sorted by USD balance and then alphabetical
+  const supportedTokens = useSelector(swappableTokensSelector)
+  const swappableTokens = useMemo(() => {
+    const tokensWithUsdPrice = supportedTokens.filter(
+      (token) => token.isCoreToken && token.usdPrice && token.usdPrice.gt(0)
```
Comment (dievazqu) [inline src/swap/SwapScreen.tsx:66]: I think `token.isCoreToken` shouldn't be here, right? We want all the tokens that have an usdPrice regardless of whether it's native or not.

Comment (kathaypacific) [reply src/swap/SwapScreen.tsx]: AH thanks, copy + paste error 😅

**Review by dievazqu — APPROVED** [review]
LGTM!

File: src/tokens/selectors.ts (hunk 1 above)
Comment (jeanregisser) [inline src/tokens/selectors.ts:127]: nit: 
```suggestion
function tokenCompareByUsdBalanceThenByName(token1: TokenBalance, token2: TokenBalance) {
```

File: src/swap/SwapScreen.tsx [hunk 3]
```diff
@@ -47,19 +43,7 @@ const DEFAULT_SWAP_AMOUNT: SwapAmount = {
...
-  const usdPriceComparison = token2UsdBalance.comparedTo(token1UsdBalance)
-  if (usdPriceComparison === 0) {
-    const token1Name = token1.name ?? 'ZZ'
-    const token2Name = token2.name ?? 'ZZ'
-    return token1Name.localeCompare(token2Name)
-  } else {
-    return usdPriceComparison
-  }
-}
+const PRICE_IMPACT_THRESHOLD = 0.04
```
[the remark this answers is not in the document]
Comment (jeanregisser) [reply src/swap/SwapScreen.tsx:46]: Good idea!

**Review by jeanregisser — APPROVED** [review]
🚀

File: src/swap/SwapScreen.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (kathaypacific) [reply src/swap/SwapScreen.tsx:46]: ah yes, i was wondering about this too. i'll put it in

**Pull request merged**